js/apps/admin-ui/src/realm-settings/user-profile/attribute/AddTranslationsDialog.tsx

```diff
@@ -0,0 +1,464 @@
+    }
+  };
+
+  const save = async () => {
+    const formData = getValues();
+
+    try {
+      const nonEmptyTranslations = formData.translations
+        .filter((translation) => translation.value.trim() !== "")
+        .map(async (translation) => {
+          try {
+            await adminClient.realms.addLocalization(
+              {
+                realm: realmName,
+                selectedLocale: translation.locale,
+                key: formData.key,
+              },
+              translation.value,
+            );
+          } catch (error) {
+            console.error(`Error saving translation for ${translation.locale}`);
+          }
+        });
+
+      await Promise.all(nonEmptyTranslations);
+
+      toggleDialog();
+    } catch (error) {
+      console.error(`Error saving translations: ${error}`);
+    }
+  };
+
+  return (
+    <Modal
+      variant={ModalVariant.medium}
+      title={t("addTranslationsModalTitle")}
+      isOpen
+      onClose={toggleDialog}
+      actions={[
+        <Button
+          key="save"
+          data-testid="saveTranslationBtn"
+          variant="primary"
+          type="submit"
+          form="add-translation"
+          isDisabled={!isValid || isWarning}
+        >
+          {t("save")}
+        </Button>,
+        <Button
+          key="cancel"
+          data-testid="cancelTranslationBtn"
+          variant="link"
+          onClick={onCancel}
+        >
+          {t("cancel")}
+        </Button>,
+      ]}
+    >
+      <Flex
+        direction={{ default: "column" }}
+        spaceItems={{ default: "spaceItemsNone" }}
+      >
+        <FlexItem>
+          <TextContent>
+            <Text component={TextVariants.p}>
+              {t("addTranslationsModalSubTitle")}{" "}
+              <strong>{t("addTranslationsModalSubTitleBolded")}</strong>
+            </Text>
+          </TextContent>
+        </FlexItem>
+        <FlexItem>
+          <Form
+            id="add-translation"
+            data-testid="addTranslationForm"
+            onSubmit={handleSubmit(save)}
+          >
+            <FormGroup
+              className="pf-u-mt-md"
+              label={t("translationKey")}
+              fieldId="kc-translation-key"
+              isRequired
+              validated={
+                errors.key ? ValidatedOptions.error : ValidatedOptions.default
+              }
+              helperTextInvalid={t("required")}
+            >
+              <Controller
+                name="key"
+                control={control}
+                rules={{
+                  validate: (value) => (value || "").length > 0,
+                }}
+                render={({ field }) => (
+                  <KeycloakTextInput
+                    id="kc-translation-key"
+                    {...field}
+                    aria-label={t("translationKey")}
+                    data-testid="translation-key"
+                  />
+                )}
+              />
+            </FormGroup>
```

Comment:
It would be nicer when you would use the component from `ui-shared` for this:
```suggestion
            <TextControl
              name="key"
              label={t("translationKey")}
              rules={{ required: { value: true, message: t("required") } }}
              className="pf-u-mt-md"
             />
```